src/backend/distributed/deparser/deparse_role_stmts.c

```diff
@@ -380,14 +391,50 @@ AppendGrantRoleStmt(StringInfo buf, GrantRoleStmt *stmt)
 	{
 #if PG_VERSION_NUM >= PG_VERSION_16
 		DefElem *opt = NULL;
+		 int opt_count = 0;
 		foreach_ptr(opt, stmt->opt)
 		{
-			if (strcmp(opt->defname, "admin") == 0)
-			{
-				appendStringInfo(buf, " WITH ADMIN OPTION");
-				break;
-			}
+			switch (opt->defname)
+        {
+            case "admin":
+                appendStringInfo(buf, " WITH ADMIN OPTION");
+                opt_count++;
+                break;
+
+			case "inherit":
+				  if (opt->arg && IsA(opt->arg, A_Const) && !((A_Const *) opt->arg)->val.val.ival)
```

Comment:
This if statement is currently not working, and a bit confusing. I recommend you check this PR https://github.com/citusdata/citus/pull/7191 on obtaining opt's value. Something like the following:
```
// for admin we only care if it's TRUE because by default it is FALSE
bool admin_option = false;
char *optval = defGetString(opt);
if (strcmp(opt->defname, "admin") == 0 &&
	parse_bool(optval, &admin_option) && admin_option)
```